samcli/commands/check/resources/TemplateResource.py

```diff
@@ -0,0 +1,22 @@
+"""
+Super class for all resources
+"""
+from samcli.lib.providers.provider import Function
+
+
+class TemplateResource:
+    def __init__(self, resource_object: Function, resource_type: str):
```

Comment:
`resource_object` should not be type of Function.
Adding more child classes of TemplateResource will break this part if they are not Function related.

Reply:
Right now it can only be of type `Function`. As I add more resources, I will add more options to what type it can be.

Reply:
Parent classes shouldn't know what child classes need.
For example, if we add a new Layer resource extending TemplateResource, we have to update TemplateResource to:
```
class TemplateResource:
    def __init__(self, resource_object: Union[Function, Layer], resource_type: str):
```
Adding new child classes should not change parent classes.

Instead, we can either make resource_object as Any and let child classes handle checking for specific types, or remove it completely and let child classes create their own resource_object class members.